notes/zrq/20210728-01-live-deploy.txt

```diff
@@ -0,0 +1,1288 @@
+    >   ....
+    >   ....
+    >   2G9BXYCKP
+    >   2FF2VTAAM
+
+
+        # Create a JSON array of interpreter bindings.
+        sed '
+            1 i \
+"interpreterBindings": {
+        s/^\(.*\)$/"\1": ["spark", "md", "sh"]/
+        $ ! s/^\(.*\)$/\1,/
+        $ a \
+},
+            ' /tmp/001.txt \
+            | tee /tmp/002.txt
+
+
+    >   "interpreterBindings": {
+    >   "2C35YU814": ["spark", "md", "sh"],
+    >   "2EZ3MQG4S": ["spark", "md", "sh"],
+    >   ....
+    >   ....
+    >   "2G9BXYCKP": ["spark", "md", "sh"],
+    >   "2FF2VTAAM": ["spark", "md", "sh"]
+    >   },
+
+
+        # Wrap our fragment as a JSON document to check
+        sed '
+            1 i \
+{
+        $ s/,//
+        $ a \
+}
+            ' /tmp/002.txt \
+        | jq '.'
+
+
+    >   {
+    >     "interpreterBindings": {
+    >       "2C35YU814": [
+    >         "spark",
+    >         "md",
+    >         "sh"
+    >       ],
+    >       ....
+    >       ....
+    >       "2FF2VTAAM": [
+    >         "spark",
+    >         "md",
+    >         "sh"
+    >       ]
+    >     }
+    >   }
+
+
+# -----------------------------------------------------
+# Truncate any existing list.
+#[user@zeppelin]
+
+        jq '
+            del(.interpreterBindings[])
+            ' \
+        /home/fedora/zeppelin/conf/interpreter.json \
+        > /tmp/003.json
+
+        sed -n '
+            /interpreterBindings/ p
+            ' /tmp/003.json
+
+    >     "interpreterBindings": {},
+
+
+# -----------------------------------------------------
+# Replace the empty list with our fragment.
+#[user@zeppelin]
+
+    # Insert our binding list into the rest of the file.
+    sed '
+        /interpreterBindings/ {
+            r /tmp/002.txt
+            d
+            }
+        ' /tmp/003.json \
+    | jq '.' \
+    > /tmp/004.json
+
+    # Run it through 'jq' to check.
+    jq '
+        .interpreterBindings
+        ' /tmp/004.json
+
+    >   {
+    >     "2C35YU814": [
+    >       "spark",
+    >       "md",
+    >       "sh"
+    >     ],
+    >     ....
+    >     ....
+    >     "2FF2VTAAM": [
+    >       "spark",
+    >       "md",
+    >       "sh"
+    >     ]
+    >   }
+
+
+# -----------------------------------------------------
+# Replace the original interpreter.json from git.
+#[user@zeppelin]
+
+    mv /home/fedora/zeppelin/conf/interpreter.json \
+       /home/fedora/zeppelin/conf/interpreter.origin
+
+    cp /tmp/004.json \
+       /home/fedora/zeppelin/conf/interpreter.json
+
+    /home/fedora/zeppelin/bin/zeppelin-daemon.sh restart
+
+    >   Zeppelin stop                                              [  OK  ]
+    >   Zeppelin start                                             [  OK  ]
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+# Create shell script functions to wrap the REST API.
+#[user@desktop]
+
+    zeppelinurl=http://zeppelin.gaia-prod.aglais.uk:8080
```

Comment:
Are you using this as an alternative to zeppelin.aglais.uk? Since this is a live deploy I assume both would be pointing to the same deploy?